packages/dai-plugin-governance/src/SpellService.js

```diff
@@ -0,0 +1,94 @@
+import { PublicService } from '@makerdao/services-core';
+import { PAUSE } from './utils/constants';
+import DsSpellAbi from '../contracts/abis/DSSpell.json';
+import padStart from 'lodash/padStart';
+import assert from 'assert';
+import contractInfo from '../contracts/contract-info.json';
+const pauseInfo = contractInfo.pause;
+import { netIdToName } from './utils/helpers';
+
+export default class SpellService extends PublicService {
+  constructor(name = 'spell') {
+    super(name, ['smartContract', 'web3']);
+    this.eta = {};
+    this.done = {};
+    this.action = {};
+    this.executionDate = {};
+  }
+
+  getDelayInSeconds() {
+    if (this.delay) return this.delay;
+    this.delay = this._pauseContract().delay();
+    return this.delay;
+  }
+
+  async getEta(spellAddress) {
```

Comment:
what does this return when eta hasn't been set yet? (`new Date(0)` ?)

Reply:
ooh good point, I updated it to throw an error if eta isn't set yet